src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1481,7 +1481,14 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
                 expression = null;
                 if (node is IdentifierNameSyntax identifier)
                 {
-                    var type = BindNativeIntegerSymbolIfAny(identifier, diagnostics);
+                    // Don't bind nameof(nint) or nameof(nuint) so that ERR_NameNotInContext is reported.
+                    // The check for ArgumentSyntax is to allow the binding for something like nameof(nint.Equals).
+                    var type = IsInsideNameof && identifier.Parent is ArgumentSyntax &&
```

Comment:
Please add a test for an actual invocation `nameof(nint)` (such that `IsInsideNameof` returns `false`).

Reply:
@jcouv Did you mean to have an actual method named "nameof"? What should be the argument type it takes so "nint" actually becomes valid? or it still cannot be valid?

Reply:
Yes, I'm suggesting to test with method named "nameof". The following code is legal and remains legal (and binds no native integer types). 
Your PR handles is correctly, but the tests don't exercise that path as far as I can tell.

```
public class C 
{
    public string nint;
    public void nameof(string x)
    {
        nameof(nint);   
    }
}
```